src/pass.h

```diff
@@ -39,8 +39,12 @@ struct PassRegistry {
   typedef std::function<Pass*()> Creator;
 
   void registerPass(const char* name, const char* description, Creator create);
+  // Register a pass that's used for internal testing. These pass do not show up
+  // in --help.
+  void
+  registerTestPass(const char* name, const char* description, Creator create);
   std::unique_ptr<Pass> createPass(std::string name);
-  std::vector<std::string> getRegisteredNames();
+  std::vector<std::string> getRegisteredNames(bool includeHidden = false);
```

Comment:
How about an enum for Hidden/Visible? Can be used on line 58 too.

Reply:
I moved the `hidden` property to `option`, so this code is gone now.. PTAL at the new code and let me know if that looks fine.